infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -211,9 +211,9 @@ private void connectAndQueryTopology() throws LogReplicationDiscoveryServiceExce
             try {
                 // Connect to Cluster Manager and Retrieve Topology Info
                 log.info("Connecting to Cluster Manager adapter...");
-                clusterManagerAdapter.connect(this);
+                clusterManagerAdapter.setCorfuReplicationDiscoveryService(this);
```

Comment:
Shouldn't we call the clusterManagerAdapter.start() at this point? so it connects to the remote topology provider? If someone implements the interface this will never be triggered.

Reply:
The clusterManager talk to the application manager to start. This is just set the DiscoveryService for callback when there is a new config.

Reply:
exactly, but I didn't see where the start is being called. I only see it in the abstract class clubbed with the setCorfuReplicationDiscoveryService. But for the case of someone directly implementing the interface, where is the start called? Maybe I missed it but I didn't see it.

Reply:
Made the changes.